config/initializers/health_checks.rb

```diff
@@ -17,7 +17,19 @@
                                       OutgoingMessage.last.created_at
                                   end
 
+  xapian_queue_limit = 500
+  xapian_queue_check = HealthChecks::Checks::LimitCheck.new(
+    limit: xapian_queue_limit,
+    failure_message: _('The number of Xapian index jobs, older than 30 ' \
+                       'minutes, is over {{limit}}', limit: xapian_queue_limit),
+    success_message: _('The number of Xapian index jobs, older than 30 ' \
+                       'minutes, is under {{limit}}', limit: xapian_queue_limit)
+  ) do
+    ActsAsXapian::ActsAsXapianJob.where(created_at: (...30.minutes.ago)).count
```

Comment:
I assume this does what it appears to - I'm not familiar with the underlying code though!